src/lib/execution_plan.rs

```diff
@@ -549,6 +550,67 @@ impl<'a> ExecutionPlanBuilder<'a> {
             };
         }
 
-        Ok(ExecutionPlan { steps })
+        Ok(ExecutionPlan {
+            steps: Self::handle_before_each_and_after_each_tasks(&config, &steps)?.into_owned(),
+        })
+    }
+
+    /// `before_each` and `after_each` interspersal for the steps vector
+    fn handle_before_each_and_after_each_tasks<'b>(
+        config: &Config,
+        steps: &'b Vec<Step>,
+    ) -> Result<Cow<'b, Vec<Step>>, CargoMakeError> {
+        let mut after_and_before_each = Vec::<Step>::with_capacity(2);
+        let mut handle_some = |name: &String| -> Result<u8, CargoMakeError> {
+            let task_config = get_normalized_task(config, name, false)?;
+            if !task_config.disabled.unwrap_or(false) {
+                after_and_before_each.push(Step {
+                    name: name.to_string(),
+                    config: task_config,
+                });
+                Ok(1)
+            } else {
+                Ok(0)
+            }
+        };
+        let has_after_each: u8 = match &config.config.after_each_task {
+            None => 0,
+            Some(after_each) => handle_some(&after_each)?,
+        };
+        let has_before_each: u8 = match &config.config.before_each_task {
+            None => 0,
+            Some(before_each) => handle_some(&before_each)?,
+        };
+        let scale: u8 = has_before_each + has_after_each;
+        if scale > 0 {
+            let before_and_after_each_len = scale as usize + 1;
+            let mut interspersed_steps =
+                Vec::<Step>::with_capacity(steps.len() * before_and_after_each_len);
+            let before_special = HashSet::from(["init", "init_task"]);
+            let end_special = HashSet::from(["end", "end_task"]);
+            interspersed_steps.extend(steps.into_iter().flat_map(|e| -> Vec<Step> {
+                let mut _steps = Vec::<Step>::with_capacity(before_and_after_each_len + 1);
+                if before_special.contains(e.name.as_str()) {
+                    if has_after_each == 1 {
+                        _steps.push(
+                            unsafe { after_and_before_each.last().unwrap_unchecked() }.to_owned(),
```

Comment:
do we really need unsafe code? seems like we are doing basic stuff. i know it will be more 'long/verbose' but... can we drop the unsafe?

Reply:
Can easily drop the `unsafe` and use `unwrap()` rather than `unwrap_unchecked()`. I added it because it's in a logic block that already confirms so it's proven safe. But sure, can replace this.